go-controller/pkg/clustermanager/default_network_cluster_controller.go

```diff
@@ -0,0 +1,400 @@
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/clustermanager/subnetallocator"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube"
+	objretry "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/retry"
+	ovntypes "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+type networkClusterControllerBase struct {
+	kube         kube.Interface
+	watchFactory *factory.WatchFactory
+	stopChan     chan struct{}
+	wg           *sync.WaitGroup
+
+	// node events factory handler
+	nodeHandler *factory.Handler
+
+	// retry framework for nodes
+	retryNodes *objretry.RetryFramework
+}
+
+// defaultNetworkClusterController is the cluster controller for the default network
+// for all the nodes. Listens to the node events and allocates subnet from the
+// default cluster subnet pool.
+type defaultNetworkClusterController struct {
+	networkClusterControllerBase
+	defaultNetworkController *NodeNetworkController
+
+	hybridOverlaySubnetAllocator *subnetallocator.HostSubnetAllocator
+}
+
+func newDefaultNetworkClusterController(ovnClient *util.OVNClusterManagerClientset, wf *factory.WatchFactory) *defaultNetworkClusterController {
+
+	kube := &kube.Kube{
+		KClient: ovnClient.KubeClient,
+	}
+
+	stopChan := make(chan struct{})
+	wg := &sync.WaitGroup{}
+
+	var hybridOverlaySubnetAllocator *subnetallocator.HostSubnetAllocator
+	if config.HybridOverlay.Enabled {
+		hybridOverlaySubnetAllocator = subnetallocator.NewHostSubnetAllocator()
+	}
+	ncm := &defaultNetworkClusterController{
+		networkClusterControllerBase: networkClusterControllerBase{
+			kube:         kube,
+			watchFactory: wf,
+			stopChan:     stopChan,
+			wg:           wg,
+		},
+		defaultNetworkController:     newNodeNetworkController(kube, wf, ovntypes.DefaultNetworkName),
+		hybridOverlaySubnetAllocator: hybridOverlaySubnetAllocator,
+	}
+
+	ncm.initRetryFramework()
+	return ncm
+}
+
+func (ncm *defaultNetworkClusterController) initRetryFramework() {
+	ncm.retryNodes = ncm.newRetryFramework(factory.NodeType)
+}
+
+// Init initializes the default network subnet allocator ranges
+// and hybrid network subnet allocator ranges if hybrid overlay is enabled.
+func (ncm *defaultNetworkClusterController) Init() error {
+	if err := ncm.defaultNetworkController.InitSubnetAllocatorRanges(config.Default.ClusterSubnets); err != nil {
+		return fmt.Errorf("failed to initialize host subnet allocator ranges: %w", err)
+	}
+
+	if config.HybridOverlay.Enabled {
+		if err := ncm.hybridOverlaySubnetAllocator.InitRanges(config.HybridOverlay.ClusterSubnets); err != nil {
+			return fmt.Errorf("failed to initialize hybrid overlay subnet allocator ranges: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (ncm *defaultNetworkClusterController) Stop() {
```

Comment:
We shouldnt need any `Stop` func imo. We should have stop chan passed in from outside and that is the mechanism to stop this component. The caller also should give us a waitgroup so it can wait til this component has finished.